Add UTXO test case 'Find_UTXO'

AgwaB opened the pull request:
details:

we have to solve #27

- [x] Test case

Commit:
Add UTXO test case 'Find_UTXO'

hihiboss reviewed (commented):
Why you divide the format of `Serialize` and `Deserialize`?

I think these might be more clear!

`func (block *Block) Serialize() ([]byte, error)` & `func (block *Block) Deserialize(blockByte []byte) error`
Or
`func Serialize(block Block) ([]byte, error)` & `func Deserialize(blockByte []byte) (Block, error)`

AgwaB commented:
@hihiboss I implemented like that because of some reason... but i modified it ! moved to `common` pkg and receive as interface to get various object...
